source/common/http/http2/codec_impl.cc

```diff
@@ -317,8 +317,8 @@ void ConnectionImpl::StreamImpl::saveHeader(HeaderString&& name, HeaderString&&
 void ConnectionImpl::StreamImpl::submitTrailers(const HeaderMap& trailers) {
   std::vector<nghttp2_nv> final_headers;
   buildHeaders(final_headers, trailers);
-  int rc =
-      nghttp2_submit_trailer(parent_.session_, stream_id_, &final_headers[0], final_headers.size());
+  int rc = nghttp2_submit_trailer(parent_.session_, stream_id_, final_headers.data(),
```

Comment:
merge master? Unrelated?

Reply:
This was failing in CI because of some scenario where `final_headers` had a size of 0, so it was causing undefined behavior. Calling `vector::data()` gives us the same information, but circumvents the undefined behavior in the empty vector case.

I have no idea how my change could have caused a regression here since I don't touch really touch anything unrelated to the new filter, so I just assumed it would be fine to do this since it's a better way of getting that pointer.

Reply:
Merge master. @antoniovicente already fixed this.